src/Security/Authentication/FrontendPreviewAuthenticator.php

```diff
@@ -68,35 +61,40 @@ public function __construct(RequestStack $requestStack, SessionInterface $sessio
     /**
      * Authenticates a front end user based on the username.
      *
-     * @param null $username
-     * 
-     * @throws \RuntimeException
+     * @param string $username
+     *
+     * @return bool
      */
-    public function authenticateFrontendUser($username = null): void
+    public function authenticateFrontendUser(string $username): bool
     {
-        if (null === $username) {
-            return;
+        if (!$this->session->isStarted()) {
+            return false;
         }
 
-        $request = $this->requestStack->getCurrentRequest();
+        $token = $this->tokenStorage->getToken();
 
-        if (null === $request) {
-            throw new \RuntimeException('The request stack did not contain a request');
+        // Check if a back end user is authenticated
+        if (null === $token || !$token->isAuthenticated()) {
+            return false;
         }
 
-        if (!$request->hasSession()) {
-            return;
-        }
+        $backendUser = $token->getUser();
 
-        $token = $this->tokenStorage->getToken();
+        if (!$backendUser instanceof BackendUser) {
+            return false;
+        }
 
-        // Check if a back end user is authenticated
-        if (null === $token || !$token->isAuthenticated()) {
-            return;
+        // Back end user does not have permission to log in front end users
+        if (!$backendUser->isAdmin && (empty($backendUser->amg) || !\is_array($backendUser->amg))) {
+            return false;
         }
 
         try {
-            $user = $this->userProvider->loadUserByUsername($username);
+            $frontendUser = $this->userProvider->loadUserByUsername($username);
+
+            if (!$frontendUser instanceof FrontendUser) {
+                throw new UsernameNotFoundException('User is not a front end user');
```

Comment:
This is redundant, because `FrontendUserProvider::loadUserByUsername()` already executes the `instanceof` check.